File: Core/Core/error_code.cpp
#include "error_code.h"

#include "interop.h"

using namespace Tsunami::Core;

ErrorCode::ErrorCode(const libtorrent::error_code& ec)
{
    error_ = new libtorrent::error_code(ec);
}

ErrorCode::~ErrorCode()
{
    delete error_;
}

System::String^ ErrorCode::message()
{
    return interop::from_std_string(error_->message());
}

int ErrorCode::value()
{
    return error_->value();
}

File: Tsunami/Models/ErrorCode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tsunami.Models
{
    public class ErrorCode
    {
        public string Message { get; set; }
        public int Value { get; set; }

        public ErrorCode() { /* nothing to do. just for serializator */ }

        public ErrorCode(Core.ErrorCode er)
        {
            Message = er.message();
            Value = er.value();
        }

    }
}
